src/compas/files/obj.py

```diff
@@ -371,69 +448,90 @@ def parse(self):
-                self.write_faces()
+            self.write_meshes()
 
     def write_header(self):
-        self.file.write("# OBJ\n")
-        self.file.write("# COMPAS\n")
-        self.file.write("# version: {}\n".format(compas.__version__))
-        self.file.write("# precision: {}\n".format(self.precision))
-        self.file.write("# V F E: {} {} {}\n".format(self.v, self.f, self.e))
+        self.file.write('# OBJ\n')
+        self.file.write('# COMPAS\n')
+        self.file.write('# version: {}\n'.format(compas.__version__))
+        self.file.write('# precision: {}\n'.format(self.precision))
+        self.file.write('# V F E: {} {} {}\n'.format(self.v, self.f, self.e))
         if self.author:
-            self.file.write("# author: {}\n".format(self.author))
+            self.file.write('# author: {}\n'.format(self.author))
         if self.email:
-            self.file.write("# email: {}\n".format(self.email))
+            self.file.write('# email: {}\n'.format(self.email))
         if self.date:
-            self.file.write("# date: {}\n".format(self.date))
-        self.file.write("\n")
+            self.file.write('# date: {}\n'.format(self.date))
+        self.file.write('\n')
+
+    def write_meshes(self):
+        for index, mesh in enumerate(self.meshes):
+            name = mesh.name
+            if name == 'Mesh':
+                name = 'Mesh {}'.format(index)
+            self.file.write('o {}\n'.format(name))
+            if self.unweld:
+                self._write_vertices_and_faces(mesh)
+            else:
+                self._write_vertices(mesh)
+                self._write_faces(mesh)
+            self._v += mesh.number_of_vertices()
 
-    def write_vertices(self):
-        for key in self.mesh.vertices():
-            x, y, z = self.mesh.vertex_coordinates(key)
+    def _write_vertices(self, mesh):
+        for key in mesh.vertices():
+            x, y, z = mesh.vertex_coordinates(key)
             self.file.write(self.vertex_tpl.format(x, y, z))
 
-    def write_faces(self):
-        key_index = self.mesh.key_index()
-        for fkey in self.mesh.faces():
-            vertices = self.mesh.face_vertices(fkey)
-            vertices = [key_index[key] + 1 for key in vertices]
-            vertices_str = " ".join([str(index) for index in vertices])
-            self.file.write("f {0}\n".format(vertices_str))
-
-    def write_vertices_and_faces(self):
-        index = 1
-        for face in self.mesh.faces():
-            vertices = self.mesh.face_vertices(face)
+    def _write_faces(self, mesh):
+        key_index = mesh.key_index()
+        for fkey in mesh.faces():
+            vertices = mesh.face_vertices(fkey)
+            vertices = [key_index[key] + self._v for key in vertices]
+            vertices_str = ' '.join([str(index) for index in vertices])
+            self.file.write('f {0}\n'.format(vertices_str))
+
+    def _write_vertices_and_faces(self, mesh):
+        index = 0 + self._v
```

Comment:
should be 1 + ?

Reply:
no because `self._v` now starts at 1 instead of 0

Reply:
cool would better to change to simply `index = self._v`  then?

Reply:
yes indeed

Reply:
hmm actually there might be a small bug related to that